prov/efa/test/efa_test_getinfo.c

```diff
@@ -0,0 +1,97 @@
+#include "efa_unit_tests.h"
+
+int __real_rxr_info_to_rxr(uint32_t version, const struct fi_info *core_info,
+			struct fi_info *info, const struct fi_info *hints);
+
+int __wrap_rxr_info_to_rxr(uint32_t version, const struct fi_info *core_info,
+			struct fi_info *info, const struct fi_info *hints)
+{
+	int use_real = mock_type(bool);
+	int ret;
+	if (use_real)
+		return __real_rxr_info_to_rxr(version, core_info, info, hints);
+
+	ret = mock_type(int);
+	return ret;
+}
+
+/* Currently unused but useful to fake for future unit tests */
+int __real_rxr_get_lower_rdm_info(uint32_t version, const char *node,
+                           const char *service, uint64_t flags,
+                           const struct util_prov *util_prov,
+                           const struct fi_info *util_hints,
+                           struct fi_info **core_info);
+
+int __wrap_rxr_get_lower_rdm_info(uint32_t version, const char *node,
+                           const char *service, uint64_t flags,
+                           const struct util_prov *util_prov,
+                           const struct fi_info *util_hints,
+                           struct fi_info **core_info)
+{
+	int use_real = mock_type(bool);
+	int ret;
+	if (use_real)
+		return __real_rxr_get_lower_rdm_info(version, node, service, flags, util_prov, util_hints, core_info);
+	ret = mock_type(int);
+	return ret;
+}
+
+void rxr_info_to_rxr_core_info_NULL()
+{
+	int ret;
+	will_return(__wrap_rxr_info_to_rxr, true);
+	ret = rxr_info_to_rxr(0, NULL, NULL, NULL);
+	assert_int_equal(ret, -FI_EINVAL);
+}
+
+void rxr_info_to_rxr_mock_test()
+{
+	int ret;
+	will_return(__wrap_rxr_info_to_rxr, false);
+	will_return(__wrap_rxr_info_to_rxr, 99);
+	ret = rxr_info_to_rxr(0, NULL, NULL, NULL);
+	assert_int_equal(ret, 99);
+}
+
+void rxr_info_to_rxr_basic()
+{
+	int ret;
+	struct fi_info *core_info = fi_allocinfo();
+	struct fi_info *info = fi_allocinfo();
+	struct fi_info *hints = fi_allocinfo();
+
+	will_return(__wrap_rxr_info_to_rxr, true);
+	ret = rxr_info_to_rxr(0, core_info, info, hints);
+	assert_int_equal(ret, 0);
+	assert_int_equal(info->caps, rxr_info.caps);
+	assert_int_equal(info->mode, rxr_info.mode);
```

Comment:
The first 2 tests does not do much, and can be ignored. I would really like to see a test case that we use mock function to return multiple EFA devices.

Reply:
@wzamazon Is this still blocking the PR from being merged?